File: objc/Project5/Project5/CardCell.m
//
//  CardCell.m
//  Project5
//
//  Created by Jinwoo Kim on 2/11/21.
//

#import "CardCell.h"

@implementation CardCell

- (void)flipTo:(NSString *)image hideContents:(BOOL)hideContents {
    [UIView transitionWithView:self duration:0.5
                       options:UIViewAnimationOptionTransitionFlipFromLeft
                    animations:^{
        self.card.image = [UIImage imageNamed:image];
        [self.contents setHidden:hideContents];
        [self.textLabel setHidden:hideContents];
    }
                    completion:nil];
}

@end

File: swift/Project5/Project5/CardCell.swift
//
//  CardCell.swift
//  Project5
//
//  Created by Jinwoo Kim on 2/10/21.
//

import UIKit

class CardCell: UICollectionViewCell {
    @IBOutlet var card: UIImageView!
    @IBOutlet var contents: UIImageView!
    @IBOutlet var textLabel: UILabel!
    
    var word = "?"
    
    func flip(to image: String, hideContents: Bool) {
        UIView.transition(with: self, duration: 0.5, options: .transitionFlipFromLeft, animations: {
            self.card.image = UIImage(named: image)
            self.contents.isHidden = hideContents
            self.textLabel.isHidden = hideContents
        })
    }
}
